conformance/lang/expressions/shift-expression/left_shift.balt

```diff
@@ -0,0 +1,3816 @@
+    // Shift amount = Before applying mask, In binary: 10111111, In decimal: -65
+    //                After applying mask, In binary: 111111, In decimal: 63
+    // Result = In binary: 10000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+    //        = In decimal: (-9223372036854775808)
+
+    io:println(a << f); // @output -1
+    // Explanation:
+    // Value to be shifted = Equivalent binary representation (int:Signed8): 11111111
+    //                     = Equivalent binary representation (basic type): 11111111 11111111 11111111 11111111 11111111 11111111 11111111 11111111
+    // Shift amount = Before applying mask, In binary: 11000000, In decimal: -64
+    //                After applying mask, In binary: 000000, In decimal: 0
+    // Result = In binary: 11111111 11111111 11111111 11111111 11111111 11111111 11111111 11111111
+    //        = In decimal: (-1)
+
+    io:println(a << a); // @output -9223372036854775808
+    // Explanation:
+    // Value to be shifted = Equivalent binary representation (int:Signed8): 11111111
+    //                     = Equivalent binary representation (basic type): 11111111 11111111 11111111 11111111 11111111 11111111 11111111 11111111
+    // Shift amount = Before applying mask, In binary: 11111111, In decimal: -1
+    //                After applying mask, In binary: 111111, In decimal: 63
+    // Result = In binary: 10000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+    //        = In decimal: (-9223372036854775808)
+
+    io:println(e << c); // @output -65
+    // Explanation:
+    // Value to be shifted = Equivalent binary representation (int:Signed8): 10111111
+    //                     = Equivalent binary representation (basic type): 11111111 11111111 11111111 11111111 11111111 11111111 11111111 10111111
+    // Shift amount = Before applying mask, In binary: 01000000, In decimal: 64
+    //                After applying mask, In binary: 000000, In decimal: 0
+    // Result = In binary: 11111111 11111111 11111111 11111111 11111111 11111111 11111111 10111111
+    //        = In decimal: (-65)
+}
+
+Test-Case: error
+Description: Test resultant type of the '<<' expression with 'int:Signed8' as the LHS operand and 'int' as the RHS
+             operand.
+Labels: decimal, DecimalNumber, int, int:Signed8, shift-expr, unary-minus
+
+function errorFunction(int:Signed8 x, int a) {
+    int:Signed8 _ = x << a; // @error expected 'int:Signed8' but found 'int'
+    decimal _ = x << -100; // @error expected 'decimal' but found 'int'
+}
+
+Test-Case: error
+Description: Test resultant type of the '<<' expression with 'int:Signed8' as the LHS operand and builtin subtypes of
+             integer as the RHS operand.
+Labels: decimal, float, int:Signed16, int:Signed32, int:Signed8, int:Unsigned16, int:Unsigned32, int:Unsigned8,
+        shift-expr
+
+function errorFunction(int:Signed8 x, int:Signed8 a, int:Unsigned8 b, int:Signed16 c, int:Unsigned16 d, int:Signed32 e,
```

Comment:
Do we have a test with two string type operands in shift expr

Reply:
Added in https://github.com/ballerina-platform/ballerina-spec/pull/953/commits/2fe94debf0c3d87472283ccb4c9b99ae3142eeb9.